### PR TITLE
[DEV 4067] Implement Spending by Recipient DUNS with Elasticsearch

### PR DESCRIPTION
**Pull request by sethstoudenmier**
**Description:**
Implemented spending by recipient duns with Elasticsearch alongside Postgres implementation.

**Technical details:**
Spending by Recipient DUNs builds off the idea of setting up routing for when we run our ETL for Elasticsearch so that we can accurately aggregate recipient information. The original routing field was `recipient_hash`, however, due to special cases such as `MULTIPLE RECIPIENTS` solely using the recipient_hash will not provide accurate results for the endpoint. This is why a `recipient_agg_key` was created to pair the recipient_hash + recipient_level with the recipient's name and unqiue_id. This gives accurate aggregations that take into those special cases. 

Some of the additional changes were formatting and then another change in attempt to solve the floating point number issue. While working on this PR it was noticed that some values returned from Elasticsearch had floating point precision issues. To get around this all values from Elasticsearch are converted to an `int` (to truncate the decimals) and then divided by 100. On Elasticsearch all values are summed as `_value * 100` to minimize the risk of inconsistency. With the current changes all values are matching their Postgres version for all Spending by Category endpoints.

**Requirements for PR merge:**

1. [X] Unit & integration tests updated
2. [X] API documentation updated
3. [ ] Necessary PR reviewers:
    - [x] Backend
    - [ ] Frontend <OPTIONAL>
    - [ ] Operations <OPTIONAL>
    - [ ] Domain Expert <OPTIONAL>
4. [X] Matview impact assessment completed
5. [X] Frontend impact assessment completed
6. [X] Data validation completed
7. [ ] Appropriate Operations ticket(s) created
8. [X] Jira Ticket [DEV-4067](https://federal-spending-transparency.atlassian.net/browse/DEV-4067):
    - [X] Link to this Pull-Request
    - [X] Performance evaluation of affected (API | Script | Download)
    - [X] Before / After data comparison

**Area for explaining above N/A when needed:**
```
```

### CODE REVIEW

File: usaspending_api/search/v2/views/spending_by_category_views/spending_by_agency_types.py [hunk 1]
```diff
@@ -35,7 +35,7 @@ def build_elasticsearch_result(self, response: dict) -> List[dict]:
 
             results.append(
                 {
-                    "amount": Decimal(bucket.get("sum_field", {"value": 0})["value"]) / Decimal("100"),
+                    "amount": int(bucket.get("sum_field", {"value": 0})["value"]) / Decimal("100"),
```
Comment (tony-sappe) [inline usaspending_api/search/v2/views/spending_by_category_views/spending_by_agency_types.py:38]: What makes int more accurate than Decimal?

Comment (sethstoudenmier) [reply usaspending_api/search/v2/views/spending_by_category_views/spending_by_agency_types.py:38]: Ran into an issue when testing locally where even though we store values as a scaled_float and multiple by 100 as we sum a value returned from Elasticsearch still had issues around floating point precision. The integer was correct but had an annoying .000001 added on to it. The solution was to truncate at the decimal.

Comment (tony-sappe) [reply usaspending_api/search/v2/views/spending_by_category_views/spending_by_agency_types.py:38]: OK, tracking now

File: usaspending_api/api_contracts/contracts/v2/search/spending_by_category/recipient_duns.md [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+        + `domestic`
+        + `foreign`
++ `recipient_locations` (optional, array[LocationObject], fixed-type)
++ `recipient_type_names`: `category_business` (optional, array[string])
++ `award_type_codes` (optional, FilterObjectAwardTypes)
++ `award_ids`: `SPE30018FLGFZ`, `SPE30018FLJFN` (optional, array[string])
+    Award IDs surrounded by double quotes (e.g. `"SPE30018FLJFN"`) will perform exact matches as opposed to the default, fuzzier full text matches.  Useful for Award IDs that contain spaces or other word delimiters.
++ `award_amounts` (optional, array[AwardAmounts], fixed-type)
++ `program_numbers`: `10.331` (optional, array[string])
++ `naics_codes`: `311812` (optional, array[string])
```
Comment (tony-sappe) [inline usaspending_api/api_contracts/contracts/v2/search/spending_by_category/recipient_duns.md:101]: `naics_codes` were modified recently to be a list of objects

**Pull request merged**